[cgal] update to 6.0

lrineau opened the pull request:
Update to [CGAL version 6.0](https://github.com/CGAL/cgal/releases/tag/v6.0). That new version has several breaking changes, but it seems nothing broke in vcpkg.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

lrineau committed:
[cgal] update to version 6.0

dg0yt commented on ports/cgal/portfile.cmake:
```diff
@@ -4,16 +4,10 @@ vcpkg_buildpath_length_warning(37)
     HEAD_REF master
 )
-
-vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    FEATURES
-        qt WITH_CGAL_Qt5
-)
```
This needs an update or replacement. There must be a strong coupling between the choice of features and Qt being enabled in the build.

If `WITH_CGAL_Qt5` is no longer set, it must also be removed from `MAYBE_UNUSED_VARIABLES` below.

dg0yt commented on ports/cgal/portfile.cmake:
```diff
@@ -4,16 +4,10 @@ vcpkg_buildpath_length_warning(37)
     REPO CGAL/cgal
-    REF v5.6.1
-    SHA512 943413bf3b94da066d47051b22f1c1b66a39a43dad865dd0e4912a9ae6508d9a490bcfe4ef7d662844e9a1ba6e552748fad785b52b2cce98133c2dfdfbfe1a4d
+    REF v6.0
```
```suggestion
    REF v${VERSION}
```

dg0yt commented on ports/cgal/vcpkg.json:
```diff
@@ -67,14 +66,9 @@
-        "qt5-script",
-        "qt5-svg",
-        "qt5-xmlpatterns"
+        "qtbase",
```
```suggestion
        {
          "name": "qtbase",
          "default-features": false
        },
```
vcpkg will still install the default features by default, but the user may opt out (CLI, top-level manifest).

lrineau committed:
fixes/improvements after [review by @dg0yt]

Thanks for the review.

[review by @dg0yt]: https://github.com/microsoft/vcpkg/pull/41300#pullrequestreview-2344624295

lrineau committed:
remove unused variable WITH_CGAL_Qt5

lrineau committed:
add a patch

patch from upstream PR https://github.com/CGAL/cgal/pull/8523

lrineau committed:
./vcpkg x-add-version --all

dg0yt commented on ports/cgal/PR-8523__CGAL_CGAL-protect_against_macro_free-GF.patch:
````diff
@@ -0,0 +1,88 @@
+From cab1117b50a322f8d99f10b11b9654d36e889548 Mon Sep 17 00:00:00 2001
+From: Laurent Rineau <laurent.rineau@cgal.org>
+Date: Mon, 7 Oct 2024 10:52:17 +0200
+Subject: [PATCH] CGAL_Core: protect against macro `free` https://github.com/CGAL/cgal/pull/8523
+
+From vcpkg CI, there is this error:
+
+```
+123456789101112131415161718192021222324252627C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1441~1.341\bin\Hostx64\x64\cl.exe   /TP -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_CHRONO_DYN_LINK -DBOOST_CHRONO_NO_LIB -DBOOST_CONTAINER_DYN_LINK -DBOOST_CONTAINER_NO_LIB -DBOOST_DATE_TIME_DYN_LINK -DBOOST_DATE_TIME_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_PROGRAM_O ...
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(76): error C2059: syntax error: 'constant'
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(76): note: the template instantiation context (the oldest one first) is
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(39): note: while compiling class template 'CORE::MemoryPool'
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(119): error C2988: unrecognizable template declaration/definition
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(119): error C2059: syntax error: 'constant'
+D:\installed\x64-windows\include\CGAL/CORE/Expr_impl.h(1217): error C2660: 'CORE::MemoryPool<CORE::ConstDoubleRep,1024>::_free_dbg': function does not take 2 arguments
+D:\installed\x64-windows\include\CGAL/CORE/MemoryPool.h(76): note: see declaration of 'CORE::MemoryPool<CORE::ConstDoubleRep,1024>::_free_dbg'
+D:\installed\x64-windows\include\CGAL/CORE/Expr_impl.h(1217): note: while trying to match the argument list '(void *, int)'
+```
+
+There is probably a macro named `free` somewhere.
+---
+ CGAL_Core/include/CGAL/CORE/ExprRep.h    | 4 ++--
+ CGAL_Core/include/CGAL/CORE/Impl.h       | 2 +-
+ CGAL_Core/include/CGAL/CORE/MemoryPool.h | 2 +-
+ CGAL_Core/include/CGAL/CORE/RealRep.h    | 2 +-
+ 4 files changed, 5 insertions(+), 5 deletions(-)
+
````
```suggestion
```
Please don't add this to this repo's history. The patch name has enough details.

dg0yt reviewed (commented):
Given that the lib is header-only now:

- I wonder if the port still needs the `vcpkg_buildpath_length_warning`.
- I suggest to start the portfile with
  ~~~cmake
  set(VCPKG_BUILD_TYPE release)  # header-only
  ~~~

Please update the PR title to correctly reference `[cgal]`. This will appear in git history.

lrineau committed:
useless now that CGAL is header-only (since version 5.0)

lrineau committed:
apply suggestions by @dg0yt

Thanks again.

Ref: https://github.com/microsoft/vcpkg/pull/41300#pullrequestreview-2353645190

lrineau committed:
new version of the patch

lrineau committed:
./vcpkg x-add-version --all --overwrite-version

No need to bump the version, because the previous build of 6.0 failed.

lrineau committed:
another version of the patch... hope this one will be complete

lrineau committed:
./vcpkg x-add-version --all --overwrite-version

No need to bump the version, because the previous build of 6.0 failed.

lrineau commented:
@WangWeiLin-MV This pull-request is no longer a draft, and now passes the CI tests. 😌
